Sources/MapLibreSwiftUI/MapView.swift

```diff
@@ -31,96 +37,118 @@ public struct MapView: UIViewRepresentable {
-        let mapView = MLNMapView(frame: .zero)
-        mapView.delegate = context.coordinator
-        context.coordinator.mapView = mapView
+        let controller = makeViewController()
+        controller.mapView.delegate = context.coordinator
+        context.coordinator.mapView = controller.mapView
 
         // Apply modifiers, suppressing camera update propagation (this messes with setting our initial camera as
         // content insets can trigger a change)
         context.coordinator.suppressCameraUpdatePropagation = true
-        applyModifiers(mapView, runUnsafe: false)
+        applyModifiers(controller, runUnsafe: false)
         context.coordinator.suppressCameraUpdatePropagation = false
 
-        mapView.locationManager = locationManager
+        controller.mapView.locationManager = locationManager
 
         switch styleSource {
         case let .url(styleURL):
-            mapView.styleURL = styleURL
+            controller.mapView.styleURL = styleURL
         }
 
-        context.coordinator.updateCamera(mapView: mapView,
+        context.coordinator.updateCamera(mapView: controller.mapView,
                                          camera: $camera.wrappedValue,
                                          animated: false)
-        mapView.locationManager = mapView.locationManager
+        controller.mapView.locationManager = controller.mapView.locationManager
 
         // Link the style loaded to the coordinator that emits the delegate event.
         context.coordinator.onStyleLoaded = onStyleLoaded
 
         // Add all gesture recognizers
         for gesture in gestures {
-            registerGesture(mapView, context, gesture: gesture)
+            registerGesture(controller.mapView, context, gesture: gesture)
         }
 
-        return mapView
+        return controller
     }
 
-    public func updateUIView(_ mapView: MLNMapView, context: Context) {
+    public func updateUIViewController(_ uiViewController: T, context: Context) {
         context.coordinator.parent = self
 
-        applyModifiers(mapView, runUnsafe: true)
+        applyModifiers(uiViewController, runUnsafe: true)
 
         // FIXME: This should be a more selective update
-        context.coordinator.updateStyleSource(styleSource, mapView: mapView)
-        context.coordinator.updateLayers(mapView: mapView)
+        context.coordinator.updateStyleSource(styleSource, mapView: uiViewController.mapView)
+        context.coordinator.updateLayers(mapView: uiViewController.mapView)
 
         // FIXME: This isn't exactly telling us if the *map* is loaded, and the docs for setCenter say it needs to be.
-        let isStyleLoaded = mapView.style != nil
+        let isStyleLoaded = uiViewController.mapView.style != nil
 
-        context.coordinator.updateCamera(mapView: mapView,
-                                         camera: $camera.wrappedValue,
-                                         animated: isStyleLoaded)
+        if cameraDisabled == false {
```

Comment:
Can you clarify the purpose/use case for this? We spent some good time enabling all of the potential use cases the underlying MLNMapView offers through the MapViewCamera, so I just want make sure there's not a way to already accomplish the same end goal. Or a bug/use case were still missing with the camera.

Reply:
When Maplibre-navigation is running, it is in control of the camera. We want to prevent Maplibre-DSL from interfering with the camera, that's why we added `cameraModifierDisabled()`

Reply:
Makes sense to me. Thanks!